yt_dlp/extractor/common.py

```diff
@@ -2622,7 +2622,9 @@ def extract_Initialization(source):
                             base_url = base_url_e.text + base_url
                             if re.match(r'^https?://', base_url):
                                 break
-                    if mpd_base_url and not re.match(r'^https?://', base_url):
+                    if mpd_base_url and base_url.startswith('/'):
+                        base_url = compat_urlparse.urljoin(mpd_base_url, base_url)
+                    elif mpd_base_url and not re.match(r'^https?://', base_url):
                         if not mpd_base_url.endswith('/') and not base_url.startswith('/'):
```

Comment:
`base_url.startswith('/')` is always False due to above if

Reply:
Not sure this is correct? It's elif, not if. The following test cases show root-relative, relative and absolute URLs:

```
import re

mpd_base_url = "http://abc.com/e/"
base_url = "/f/g" # root-relative -- prints 'a'
#base_url = "h/i" # relative -- prints 'b'
#base_url = "http://xyz.com/j" # absolute --prints nothing, as expected

if mpd_base_url and base_url.startswith('/'):
    print("a")
elif mpd_base_url and not re.match(r'^https?://', base_url):
    print("b")
```

Reply:
I am talking about this line `if not mpd_base_url.endswith('/') and not base_url.startswith('/'):
 `

Reply:
Right, I see what you mean. Would you like me to clean that up? I personally left that code all the code related to deal with normal-relative URLs the same.

Reply:
> Would you like me to clean that up? 

The change needed to it is a direct consequence of your code. So yes

Reply:
This should now be resolved.